jujugui/static/gui/src/app/app.js

```diff
@@ -1821,6 +1752,44 @@ YUI.add('juju-gui', function(Y) {
       }
     },
 
+    _displayShortcutsModal: function() {
+      ReactDOM.render(
+        <window.juju.components.ModalShortcuts
+          closeModal={this._clearShortcutsModal.bind(this)}
+          keybindings={this.keybindings} />,
+        document.getElementById('modal-shortcuts'));
+    },
+
+    _displaySettingsModal: function() {
+      ReactDOM.render(
+        <window.juju.components.ModalGUISettings
+          closeModal={this._clearSettingsModal.bind(this)}
+          disableCookie={
```

Comment:
Why are these passed in as well as access to the localStorage? Should the component not just resolve these internally?